hbase-client/src/main/java/org/apache/hadoop/hbase/util/PoolMap.java

```diff
@@ -45,177 +36,138 @@
  * key. A size of {@link Integer#MAX_VALUE} is interpreted as an unbounded pool.
  * </p>
  *
+ * <p>
+ * PoolMap is thread-safe. It does not remove elements automatically. Unused resources
+ * must be closed and removed explicitly.
+ * </p>
```

Comment:
now that this is thread-safe can we remove the synchronization around the instance that's in `AbstractRpcClient`?

Reply:
No or at least not trivial. Synchronization is used for stopping also and we would need some extra methods in `PoolMap` like `close()`, `drainTo()`, etc.